File: webconsole/src/main/java/com/bytex/snamp/web/serviceModel/charts/ScaleInChart.java
package com.bytex.snamp.web.serviceModel.charts;

import com.bytex.snamp.supervision.elasticity.ScalingMetrics;
import org.codehaus.jackson.annotate.JsonTypeName;

/**
 * Represents rate of downscale.
 * @author Roman Sakno
 * @version 2.1
 * @since 2.0
 */
@JsonTypeName("scaleIn")
public final class ScaleInChart extends ScalingRateChart {
    public ScaleInChart() {
        super(ScalingMetrics::scaleIn);
    }

    @Override
    protected NumericAxis createAxisY() {
        final NumericAxis axis = new NumericAxis();
        axis.setUOM("operations");
        axis.setName("downscale");
        return axis;
    }
}

File: webconsole/src/main/resources/_webapp/src/app/charts/model/charts/scale.in.chart.ts
import { ScalingRateChart } from "../scaling.rate.chart";
import { NumericAxis } from "../axis/numeric.axis";
import { AbstractChart } from "../abstract.chart";

export class ScaleInChart extends ScalingRateChart {
    get type():string {
        return AbstractChart.SCALE_IN;
    }

    public createDefaultAxisY() {
        let _axis:NumericAxis = super.createDefaultAxisY();
        _axis.unitOfMeasurement = "operations";
        _axis.name = "downscale";
        return _axis;
    }
}
